dataclients/kubernetes/clusterstate.go

```diff
@@ -16,6 +16,7 @@ type clusterState struct {
 	services        map[definitions.ResourceID]*service
 	endpoints       map[definitions.ResourceID]*endpoint
 	cachedEndpoints map[endpointID][]string
+	secrets         []*secret
```

Comment:
I think it should hold unique value per namespace+name similar to services and endpoints, e.g. `secrets map[definitions.ResourceID]*secret`